src/IGOVesting.sol

```diff
@@ -235,25 +209,14 @@ contract IGOVesting is Ownable, Initializable, IIGOVesting {
 
     function claimDistribution(
         address _wallet
-    ) public override returns (bool) {
-        uint256 idx = vestingPool.hasWhitelist[_wallet].arrIdx;
-        WhitelistInfo storage whitelist = vestingPool.whitelistPool[idx];
-
-        require(whitelist.amount != 0, "user already refunded");
+    ) public override returns (bool, uint256) {
```

Comment:
This was deemed as a vulnerability as anyone can claim on behalf of anyone and deny their ability to refund. I've made a fix on review2 PR, would be great if you could follow the same or make an internal function just the way the refund was refactored